lib/WorseReflection/Core/Inference/Resolver/ObjectCreationExpressionResolver.php

```diff
@@ -27,12 +28,13 @@ public function __construct(private GenericMapResolver $resolver)
     public function resolve(NodeContextResolver $resolver, Frame $frame, Node $node): NodeContext
     {
         assert($node instanceof ObjectCreationExpression);
-        if (false === $node->classTypeDesignator instanceof Node) {
-            throw new CouldNotResolveNode(sprintf('Could not create object from "%s"', get_class($node)));
-        }
 
+        $className = $node->classTypeDesignator;
+        if (false === $className instanceof Node) {
```

Comment:
can this be explicit? in what case should it be an anonymous class?

Reply:
I don't think so. From what I have seen is that in the case of an anonymous class it is a Token pointing to something. (When I checked it it was something inside the anon class :shrug: ) As I see it there is no good way to detect an anonymous class. But that's a parser issue.

Reply:
so at least if it's a Token?

Reply:
Yes. But I don't know if that's always the case.

Reply:
if it's the case so far then let's go with that :)

Reply:
Replaced it with a check for `Token`